watertap/ui/fsapi.py

```diff
@@ -305,6 +321,176 @@ def add(self, *args, data: Union[dict, ModelExport] = None, **kwargs) -> object:
         self.model_objects[key] = model_export
         return model_export
 
+    def from_csv(self, file: Union[str, Path], flowsheet):
+        """Load multiple exports from the given CSV file.
+
+        CSV file format rules:
+
+            * Always use a header row. The names are case-insensitive, order is
+              not important. The 'name', 'obj', and 'ui_units' columns are required.
+            * Columns names should match the non-computed names in :class:`ModelExport`.
+              See `.add()` for a list.
+            * The object to export should be in a column named 'obj', prefixed with 'fs.'
+            * For units, use Pyomo units module as 'units', e.g., 'mg/L' is `units.mg / units.L`
+
+        For example::
+
+            name,obj,description,ui_units,display_units,rounding,is_input,input_category,is_output,output_category
+            Leach liquid feed rate,fs.leach_liquid_feed.flow_vol[0],Leach liquid feed volumetric flow rate,units.L/units.hour,L/h,2,TRUE,Liquid feed,FALSE,
+            Leach liquid feed H,"fs.leach_liquid_feed.conc_mass_comp[0,'H']",Leach liquid feed hydrogen mass composition,units.mg/units.L,mg/L,3,TRUE,Liquid feed,FALSE,
+            .......etc.......
+
+        Args:
+            file: Filename or path. If not an absolute path, start from the
+                  directory of the caller's file.
+            flowsheet: Flowsheet used to evaluate the exported objects.
+
+        Returns:
+            int: Number of exports added
+
+        Raises:
+            IOError: if input file doesn't exist
+            ValueError: Invalid data in input file (error message will have details)
+        """
+        _log.debug(f"exports.add: from csv filename={file}")
+
+        # compute path
+        path = Path(file) if not isinstance(file, Path) else file
+        if path.is_absolute():
+            _log.debug(
+                f"Reading CSV data for interface exports from " f"absolute path: {path}"
+            )
+            text = open(path, "r", encoding="utf-8").read()
+        else:
+            caller = inspect.getouterframes(inspect.currentframe())[1]
+            caller_mod = inspect.getmodule(caller.frame).__name__
+            if "." not in caller_mod:  # not in a package
+                path = Path(caller.filename).parent / file
+                text = open(path, "r", encoding="utf-8").read()
+            else:
+                caller_pkg = ".".join(caller_mod.split(".")[:-1])  # strip module
+                _log.debug(
+                    f"Reading CSV data for interface exports from: "
+                    f"file={path}, module={caller_mod}, package={caller_pkg}"
+                )
+                try:
+                    text = files(caller_pkg).joinpath(path).read_text()
```

Comment:
Some of this could probably be simplified by utilizing `pyomo.common.fileutils.this_file_dir`. https://github.com/Pyomo/pyomo/blob/main/pyomo/common/fileutils.py#L71

Reply:
Since that contains only part of the logic I need, I'd prefer to keep it all here